### DISCUSSION TITLE
Text

### DISCUSSION THREAD

**Discussion by mdorkhah** [GateNLP/python-gatenlp#99] [Ideas]
Why Text does not accept Type argument?

class Text (text, name=None, matchcase=True)

I need to check within a specific type for a compiled regular expression.

**Comment by johann-petrak** [comment]
The Text parser describes just the text to match.

In order to match ANY pattern within any annotation that has a given type, the within method of that pattern can be used, e.g. `pattern = Text(re.compile(r"this|that").within(type="SomeType")`

The same works to match one type within another: `pattern = AnnAt(type="typeA").within(type="typeB")` or even to match a whole sequence or other complex pattern within an annotation.

**Reply by mdorkhah** [reply]
Noted with thanks.
